R/python.R

```diff
@@ -231,41 +250,55 @@ py_has_convert <- function(x) {
   if (py_is_null_xptr(x) || !py_available())
     return(NULL)
 
-  # deterimine whether this object converts to python
-  convert <- py_has_convert(x)
-
   # special handling for embedded modules (which don't always show
   # up as "attributes")
   if (py_is_module(x) && !py_has_attr(x, name)) {
-    module <- py_get_submodule(x, name, convert)
+    module <- py_get_submodule(x, name, py_has_convert(x))
     if (!is.null(module))
       return(module)
   }
 
-  # get the attrib
-  if (is.numeric(name) && (length(name) == 1) && py_has_attr(x, "__getitem__"))
-    attrib <- x$`__getitem__`(as.integer(name))
-  else if (inherits(x, "python.builtin.dict"))
-    attrib <- py_dict_get_item(x, name)
-  else
-    attrib <- py_get_attr(x, name)
+  # get the attrib and convert as needed
+  if (prefer_attr) {
 
-  # convert
-  if (convert || py_is_callable(attrib)) {
+    if (py_has_attr(x, name)) {
```

Comment:
These checks should only occur if the user is passing in a scalar string (length one character vector).